src/Filesystem/PathResolver.php

```diff
@@ -0,0 +1,196 @@
+<?php namespace October\Rain\Filesystem;
+
+use October\Rain\Exception\ApplicationException;
+
+/**
+ * A utility to resolve paths to their canonical location and handle path queries.
+ *
+ * @package october\filesystem
+ * @author Ben Thomson
+ */
+class PathResolver
+{
+    /** @var string The path to resolve or compare. */
+    protected $path;
```

Comment:
I don't think this is worth being a class that needs to be instantiated, each method could very easily just be a static method, no need for state here.

Reply:
Yeah good point, I've simplified the class a bit.